server/tserver/src/main/java/org/apache/accumulo/tserver/scan/NextBatchTask.java

```diff
@@ -48,6 +49,18 @@ public NextBatchTask(TabletHostingServer server, long scanID, AtomicBoolean inte
     }
   }
 
+  private void recordException(SingleScanSession scanSession) {
+    if (scanSession != null && server.getScanMetrics() != null) {
+      String executorName = getExecutorName(scanSession);
+      server.getScanMetrics().incrementExecutorExceptions(executorName);
+    }
+  }
+
+  private String getExecutorName(SingleScanSession scanSession) {
+    String executorName = scanSession.getExecutionHints().get("scan_type");
```

Comment:
The scan hint can influence which executor is used, but it does not have to be present and table config can also influence what scan executor is used.  Should probably do the following to get the executor name.

```suggestion
    return scanSession.scanParams.getScanDispatch().getExecutorName();
```